src/components/calculator.js

```diff
@@ -0,0 +1,52 @@
+import PropTypes from 'prop-types';
+
+const Calculator = () => (
+  <div>
+    <Grid />
+  </div>
+);
+
+const calculatorSymbols = [
+  'AC',
+  '+/-',
+  '%',
+  '÷',
+  '7',
+  '8',
+  '9',
+  'x',
+  '4',
+  '5',
+  '6',
+  '-',
+  '1',
+  '2',
+  '3',
+  '+',
+  '0',
+  '.',
+  '-',
+];
+
+const Grid = () => (
+  <div className="border" id="grid">
+    <Display />
+    {calculatorSymbols.map((symbol, index) => {
+      const id = `id${index}`;
+      return <Buttons symbol={symbol} key={symbol} id={id} />;
+    })}
+  </div>
+);
+
+const Display = () => (
+  <div className="border" id="display">
+    <h2>0</h2>
+  </div>
+);
+
+const Buttons = ({ symbol, id }) => <button id={id} type="button">{symbol}</button>;
+Buttons.propTypes = {
+  symbol: PropTypes.string.isRequired,
+  id: PropTypes.number.isRequired,
+};
```

Comment:
- Great work dividing the logic into multiple components. However, it's cleaner, more readable, and optimized if you divided each component into a separate file. 😉

Reply:
@MahmoudBakr23, thank you so much for the feedback. I think your idea is great and I will implement it right away!